packages/ui/components/data-table/filters/index.tsx

```diff
@@ -41,6 +41,9 @@ function ColumnVisibilityButtonComponent<TData>(
   const { t } = useLocale();
   const allColumns = table.getAllLeafColumns();
   const [open, setOpen] = useState(false);
+  const [visibleColumns, setVisibleColumns] = useState(
```

Comment:
Track state internally since it lives outside we would need to force a re-rrender